Firestore/Source/Core/FSTFirestoreClient.m

```diff
@@ -287,27 +288,23 @@ - (void)getDocumentsFromLocalCache:(FIRQuery *)query
                         completion:(void (^)(FIRQuerySnapshot *_Nullable query,
                                              NSError *_Nullable error))completion {
   [self.workerDispatchQueue dispatchAsync:^{
+
     FSTDocumentDictionary *docs = [self.localStore executeQuery:query.query];
+    FSTDocumentKeySet *remoteKeys = [FSTDocumentKeySet keySet];
 
-    __block FSTDocumentSet *documents =
-        [FSTDocumentSet documentSetWithComparator:query.query.comparator];
-    FSTDocumentSet *oldDocuments = documents;
-    [docs enumerateKeysAndObjectsUsingBlock:^(FSTDocumentKey *key, FSTDocument *value, BOOL *stop) {
-      documents = [documents documentSetByAddingDocument:value];
-    }];
+    FSTView *view = [[FSTView alloc] initWithQuery:query.query remoteDocuments:remoteKeys];
+    FSTViewDocumentChanges *viewDocChanges = [view computeChangesWithDocuments:docs];
+    FSTViewChange *viewChange = [view applyChangesToDocuments:viewDocChanges];
+    FSTAssert(viewChange.limboChanges.count == 0,
+              @"View returned limbo docs before target ack from the server.");
```

Comment:
This assertion text makes no sense in this context (there won't be a target ack from the server since we're not starting a listen with the server).

Rather this should be `@"View returned limbo documents during local-only query execution"`.

Reply:
Done.